forge-game/src/main/java/forge/game/phase/PhaseHandler.java

```diff
@@ -1264,4 +1266,33 @@ public int getExtraTurnForPlayer(final Player p) {
         }
         return count;
     }
+
+    private void handleMultiplayerEffects() {
+        // CR 800.4m When a player leaves the game, any continuous effects with durations that last until that
+        // player’s next turn or until a specific point in that turn will last until that turn would have begun
+        int oldPlayerIdx = game.getRegisteredPlayers().indexOf(playerPreviousTurn);
+        final int playerIdx = game.getRegisteredPlayers().indexOf(playerTurn);
+        final int direction = game.getTurnOrder().getShift();
+        while (oldPlayerIdx != playerIdx) {
+            oldPlayerIdx += direction;
+            if (oldPlayerIdx < 0) {
+                oldPlayerIdx = game.getRegisteredPlayers().size() - 1;
+            } else if (oldPlayerIdx > game.getRegisteredPlayers().size() - 1) {
+                oldPlayerIdx = 0;
+            }
+            Player p = game.getRegisteredPlayers().get(oldPlayerIdx);
+            if (p.hasLost()) {
+                // CR 702.26n
+                Untap.doPhasing(p);
```

Comment:
Will this phase out creatures with Phasing?

Reply:
The function could do that for the AP, but since it's called for a player who's no longer around nothing will be there

It will phase in those "phased out under the control of a player who has left the game [...] after that player's next turn would have begun"

To set up such a scenario you could:
- steal the creature with a temporary control change effect
- phase it out
- proceed to lose
- the control effect ends, the creature changes to the previous controller's battlefield (while still phased out)

I have to note the timing is slightly off because it should phase in with all the others but the cases where that matters will be even rarer
So for now I feel more comfortable with keeping this rather close together